### ISSUE TITLE
Out of tree builds are impossible with config.h in source dir

### ISSUE THREAD

**Issue by phirestalker** [ozonesecurity/ozonebase#154]
I noticed this error while compiling and it shows up in your automated compiles as well. Is this going to cause any issues for the nvrcli sample or any custom apps?

`[ 65%] Performing configure step for 'ffmpeg'
cd /home/travis/build/ozonesecurity/ozonebase/ffmpeg-prefix/src/ffmpeg-build && PKG_CONFIG_PATH=/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/lib/pkgconfig /home/travis/build/ozonesecurity/ozonebase/externals/ffmpeg/configure --enable-shared --libdir=/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/lib --prefix=/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot --enable-libopenh264 --extra-ldflags=-L/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/lib --extra-cflags=-I/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/include/wels --pkg-config-flags=--define-variable=libdir=/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/lib --pkg-config-flags=--define-variable=includedir=/home/travis/build/ozonesecurity/ozonebase/buildroot/home/travis/ozoneroot/include
Out of tree builds are impossible with config.h in source dir.
If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "config.log" produced by configure as this will help
solve the problem.
make[2]: *** [ffmpeg-prefix/src/ffmpeg-stamp/ffmpeg-configure] Error 1
make[2]: Leaving directory `/home/travis/build/ozonesecurity/ozonebase'
make[1]: *** [CMakeFiles/ffmpeg.dir/all] Error 2
make[1]: Leaving directory `/home/travis/build/ozonesecurity/ozonebase'
make: *** [all] Error 2`

**Comment by pliablepixels** [comment]
I don't see this whilst using the build script `./ozone-build.sh` (I haven't applied your patches). How exactly are you building?

**Comment by phirestalker** [comment]
I saw it in the automated builds. Doesn't the automated travisci use the build script? I am using the ./ozone-build.sh script as well. You have to scroll through, this is not at the bottom as it is for one of the external builds I believe.

**Comment by pliablepixels** [comment]
Yes, travis also uses `./ozone-build.sh` [ref here](https://github.com/ozonesecurity/ozonebase/blob/master/.travis.yml#L65). Maybe something that changed in travis - will check

**Comment by pliablepixels** [comment]
I just did a test build on travis [here](https://travis-ci.org/ozonesecurity/ozonebase/builds/412767301?utm_source=github_status&utm_medium=notification) and did not experience any errors. Maybe that error was transient? In the past I've noticed travis VMs randomly failed at some point or the other.

**Comment by phirestalker** [comment]
from your link, it is here. 
![screenshot 2018-08-06 13 47 56](https://user-images.githubusercontent.com/7477095/43740271-a2fee8f4-997f-11e8-9230-e77d6836e3ec.png)

**Comment by pliablepixels** [comment]
Ok thanks - we can ignore this for now. Doesn't affect our build process.
